CONTRIBUTING.md

```diff
@@ -543,10 +533,47 @@ Refer datalad/config.py for information on how to add these environment variable
 - *DATALAD_ALLOW_FAIL*:
   Instructs `@never_fail` decorator to allow to fail, e.g. to ease debugging.
 
-# Changelog section
+# Release(s) workflow
+
+## Branches
+
+- `master`: changes toward the next `MAJOR.MINOR.0` release (e.g., `0.13` as of time of writing).
+  It is also used for release candidates (with `rcX` suffix)
+- `maint`: bug fixes and minor enhancements for the latest released `MAJOR.MINOR.PATCH`
+ (e.g., `0.12.7` as of time of writing)
+- `MAJOR.MINOR`: generally not used, unless some bugfix release with a crticial bugfix is needed.
+
+## Workflow
+
+- upon release of `MAJOR.MINOR.0`, `maint` branch need to be fast-forwarded to that release
+- bug fixes to functionality released within the `maint` branch, should (HIGHLY RECOMMENDED) be
+  submitted against `maint ` branch
+- cherry-picking fixes from `master` into `maint` is allowed where needed
+- `master` branch accepts PRs with new functionality
+- `master` branch merges `maint` as frequently as possible/needed 
```

Comment:
"as possible/needed" seems to be presenting two conflicting options.  FWIW my handling and preference for it is more towards the "needed".